source/common/event/file_event_impl.cc

```diff
@@ -85,9 +85,6 @@ void FileEventImpl::assignEvents(uint32_t events, event_base* base) {
 
 void FileEventImpl::updateEvents(uint32_t events) {
   ASSERT(dispatcher_.isThreadSafe());
```

Comment:
I'd like either a comment here on why this is important, or a comment in the test calling out that it's regression testing [info from PR description] just to ensure no clever person decides to improve perf by undoing this PR :-P

Reply:
Comment added.  Thanks for asking for further info in the code, there's a lot of subtle behavior that can be accidentally missed.